### PR TITLE
[internal]: pkg/params: add more explicit validation checks and unit tests

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**

### CODE REVIEW

File: pkg/params/params.go [hunk 1]
```diff
@@ -91,7 +90,10 @@ func (p *ParamDesc) GetTitle() string {
 }
 
 func (p *ParamDesc) ToParam() *Param {
-	return &Param{ParamDesc: p}
+	return &Param{
+		ParamDesc: p,
+		value:     p.DefaultValue,
```
Comment (mqasimsarfraz) [inline pkg/params/params.go:95]: Nice. This fixes the issue I was hitting regarding default value not being set [here](https://github.com/inspektor-gadget/inspektor-gadget/blob/michael/gadget-registry-local/pkg/gadgets/top/ebpf/tracer/tracer.go#L442) causing panics!

Comment (mauriciovasquezbernal) [reply pkg/params/params.go:95]: Oh yes: I think it's the same as https://github.com/inspektor-gadget/inspektor-gadget/pull/1281/files#r1089072222

Comment (mqasimsarfraz) [reply pkg/params/params.go:95]: Just tested it, `runtimes` flags are working fine as well!

File: pkg/params/params_test.go [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+		},
+		{
+			name:     "Uint64()",
+			value:    "2222222222",
+			expected: uint64(2222222222),
+			getter:   func(p *Param) any { return p.AsUint64() },
+		},
+		{
+			name:     "String()",
+			value:    "eW91J3JlIGN1cmlvdXM=",
```
Comment (mqasimsarfraz) [inline pkg/params/params_test.go:107]: :laughing:

File: pkg/params/validator_test.go [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+			bitSize:       bitSize,
+			value:         strconv.FormatInt(-(1<<(bitSize-1))-1, 10),
+			expectedError: true,
+			validateFn:    ValidateInt,
+		})
+	}
+
+	// ValidateUint
+	for _, bitSize := range bitSizes {
+		// test max and min numbers allowed
```
Comment (mqasimsarfraz) [inline pkg/params/validator_test.go:97]:
```suggestion
		// test max number allowed
```

**Review by mqasimsarfraz — APPROVED** [review]

File: pkg/params/validators.go [hunk 4]
```diff
@@ -23,20 +23,56 @@ import (
...
+	TypeUint32 TypeHint = "uint32"
+	TypeUint64 TypeHint = "uint64"
 )
 
 var typeHintValidators = map[TypeHint]ParamValidator{
-	TypeInt:  ValidateNumber, // TODO: more specific
-	TypeUint: ValidateNumber, // TODO: more specific
-	TypeBool: ValidateBool,
+	TypeBool:   ValidateBool,
+	TypeInt:    ValidateInt(strconv.IntSize),
```
Comment (flyth) [inline pkg/params/validators.go:42]: I'm not totally sure whether we should use `IntSize` here - maybe it's better to stick with a hardcoded 64 in that case. Reason being: if used in a 32 bit kubectl-gadget binary, this would be different to what runs on the server and be too strict. It's a bit edge casey, I admit ;)

**Review by flyth — APPROVED** [review]
Awesome, thanks! Those tests are really comprehensive!

LGTM

File: pkg/params/validators.go (hunk 4 above)
Comment (mauriciovasquezbernal) [reply pkg/params/validators.go:42]: I had a offline talk with @flyth, the plan is to remove the TypeInt / TypeUint as those types with a specific size are difficult to handle in distributed systems as client and server could be compiled on different architectures. I'll merge this PR and those will be removed in https://github.com/inspektor-gadget/inspektor-gadget/pull/1281.
